### PR TITLE
✨♻️ Rewrite backend using fastapi

### PR DESCRIPTION
**Pull request by pajowu**

### CODE REVIEW

**Review by anuejn — CHANGES_REQUESTED** [review]

**Comment by anuejn** [comment]
Nice Work so far

File: backend/__pypackages__/.gitignore [hunk 1]
```diff
@@ -1,5 +1,2 @@
-# We need to keep the __pypackages__ directory to make pdm use PEP 582 mode and not create a venv.
```
[the remark this answers is not in the document]
Comment (pajowu) [reply backend/__pypackages__/.gitignore:1]: This wasn't on purpose. I just recreated the entire backend directory, deleting everything including the file with this comment. I'll add it again

File: frontend/package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "check:eslint": "eslint --max-warnings 0 --ext .ts,.tsx .",
     "check:format": "prettier --check 'src/**/*.(ts|tsx)'",
     "test:fast": "echo never mind for now. these tests will be executed in pre-commit",
-    "generate-openapi": "openapi-typescript ../backend/openapi-schema.yml --output src/openapi-schema.ts"
+    "generate-openapi": "openapi-typescript ../backend/openapi-schema.json --output src/openapi-schema.ts && node scripts/trim_whitespace.mjs src/openapi-schema.ts"
```
[the remark this answers is not in the document]
Comment (pajowu) [reply frontend/package.json:14]: I wasn't sure whether we want a dependency on pyyaml, but 🤷

File: backend/scripts/new_user.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--user", required=True)
+    parser.add_argument("--pass", required=True)
+    args = parser.parse_args()
+    session = next(get_session())
+    try:
+        user = create_user(
+            session=session, username=args.user, password=getattr(args, "pass")
```
[the remark this answers is not in the document]
Comment (pajowu) [reply backend/scripts/new_user.py:15]: `pass` is a reserved keyword => `args.pass` is invalid code

**Review by anuejn — APPROVED** [review]
very, very cool

**Pull request merged**